test/options/identifierCase.ts

```diff
@@ -0,0 +1,71 @@
+import dedent from 'dedent-js';
+
+import { FormatFn } from '../../src/sqlFormatter.js';
+
+export default function supportsIdentifierCase(format: FormatFn) {
```

Comment:
There are likely more tests needed here to cover:

- that all identifiers in expression like `foo.bar.baz` get uppercased,
- variables
- parameters
- array names (treated differently by lexer)
- various types of identifier quoting styles

As most of these things depend on the dialect, it might be better to just add additional tests to places like `supportsIdentifiers()`.

Reply:
My change is only for unquoted identifiers - I think the other tokens types should not be included in this change.

Reply:
I added a test for multi-part identifiers in commit [558efa3](https://github.com/sql-formatter-org/sql-formatter/pull/663/commits/558efa36bfa434fdb672408a2fed5ccaaa4d0359)